code/modules/clothing/accessories/holomap.dm

```diff
@@ -221,8 +247,65 @@ var/list/holomap_cache = list()
 			animate(alpha = 255, time = 2, easing = SINE_EASING)
 			holomap_images += I
 
+	//Additional things we might want to track
+	extra_update()
+
 	activator.client.images |= holomap_images
 
+/obj/item/clothing/accessory/holomap_chip/proc/extra_update()
+	return
+
+/obj/item/clothing/accessory/holomap_chip/deathsquad/extra_update()
+	var/turf/T = get_turf(src)
+	for(var/obj/mecha/combat/marauder/maraud in mechas_list)
+		if(!istype(maraud,/obj/mecha/combat/marauder/series) && !istype(maraud,/obj/mecha/combat/marauder/mauler) && (T.z == maraud.z))//ignore custom-built and syndicate ones
+			var/i = 0
+			if(maraud.occupant)
+				i++
+
+			var/holomap_marker = "marker_\ref[src]_\ref[maraud]_[i]"
+
+			if(!(holomap_marker in holomap_cache))
+				var/pref = "mar"
+				if (istype(maraud,/obj/mecha/combat/marauder/seraph))
+					pref = "ser"
+				holomap_cache[holomap_marker] = image('icons/holomap_markers.dmi',"[pref][i]")
+
+			var/image/I = holomap_cache[holomap_marker]
+			I.plane = HUD_PLANE
+			I.loc = activator.hud_used.holomap_obj
+
+			if(maraud.occupant)
+				I.layer = HUD_ABOVE_ITEM_LAYER
+			else
+				I.layer = HUD_ITEM_LAYER
+
+			if(!I.pixel_x || !I.pixel_y)
+				I.pixel_x = maraud.x - T.x + activator.client.view*WORLD_ICON_SIZE + 8*(WORLD_ICON_SIZE/32)
+				I.pixel_y = maraud.y - T.y + activator.client.view*WORLD_ICON_SIZE + 9*(WORLD_ICON_SIZE/32)
+
+			animate(I,alpha = 255, pixel_x = maraud.x - T.x + activator.client.view*WORLD_ICON_SIZE + 8*(WORLD_ICON_SIZE/32), pixel_y = maraud.y - T.y + activator.client.view*WORLD_ICON_SIZE + 9*(WORLD_ICON_SIZE/32), time = 5, loop = -1, easing = LINEAR_EASING)
+			animate(alpha = 255, time = 8, loop = -1, easing = SINE_EASING)
+			animate(alpha = 0, time = 5, easing = SINE_EASING)
+			animate(alpha = 255, time = 2, easing = SINE_EASING)
+			holomap_images += I
+
+
+/obj/item/clothing/accessory/holomap_chip/proc/update_marker()
+	return
+
+/obj/item/clothing/accessory/holomap_chip/ert/update_marker()
+	marker_prefix = "ert"
+	var/obj/item/clothing/under/U = attached_to
+	if(U && ishuman(U.loc))
+		var/mob/living/carbon/human/H = U.loc
+		var/obj/item/helmet = H.get_item_by_slot(slot_head)
+		if(helmet)
```

Comment:
Then you can check if the path exists in the list and access it directly using `[]` instead of using a for counter loop.